merkle/log_verifier.go

```diff
@@ -212,3 +213,35 @@ func (v LogVerifier) VerifyConsistencyProof(snapshot1, snapshot2 int64, root1, r
 
 	return nil // Proof OK.
 }
+
+// VerifiedPrefixHashFromInclusionProof calculates a root hash over leaves
+// [0..index), based on the inclusion |proof| and |leafHash| for a leaf at the
+// specified |index| in a tree of the specified |size| with the passed in
+// |root| hash.
+// Returns an error if the |proof| verification fails. The resulting smaller
+// tree's root hash is trusted iff the bigger tree's |root| hash is trusted.
+func (v LogVerifier) VerifiedPrefixHashFromInclusionProof(
+	index, size int64,
+	proof [][]byte, root []byte, leafHash []byte,
+) ([]byte, error) {
+	if err := v.VerifyInclusionProof(index, size, proof, root, leafHash); err != nil {
+		return nil, err
+	}
+	if index == 0 {
```

Comment:
Do you want this above the call to `VerifyInclusionProof`?

Reply:
Also `VerifyInclusionProof` conveniently does some sanity checking of `index` and `size` params.

Reply:
The danger of relying on other code for sanity checking is if it gets refactored the checks can get dropped without realizing it. Don't think that's a big issue here but it can be.